src/operator/tensor/dot-inl.h

```diff
@@ -443,46 +444,22 @@ struct DotCsrRspDnsByRowBlocks {
                                   const nnvm::dim_t nnr_r,
                                   const nnvm::dim_t num_rows,
                                   const nnvm::dim_t num_cols,
-                                  const nnvm::dim_t seg_len) {
+                                  const nnvm::dim_t seg_len,
+                                  const std::unordered_map<RType, nnvm::dim_t>* row_idx_map) {
```

Comment:
Passing by const reference is preferable in C++ if you don't change the content.